src/chart/generateCategoricalChart.tsx

```diff
@@ -1283,35 +1274,20 @@ export const generateCategoricalChart = ({
       return null;
     };
 
-    componentDidUpdate(prevProps: CategoricalChartProps) {
-      // add syncId
-      if (_.isNil(prevProps.syncId) && !_.isNil(this.props.syncId)) {
-        this.addListener();
-      }
-      // remove syncId
-      if (!_.isNil(prevProps.syncId) && _.isNil(this.props.syncId)) {
-        this.removeListener();
-      }
-    }
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    componentDidUpdate() {}
 
     componentWillUnmount() {
-      this.clearDefer();
-      if (!_.isNil(this.props.syncId)) {
-        this.removeListener();
-      }
-      this.cancelThrottledTriggerAfterMouseMove();
-    }
-
-    cancelThrottledTriggerAfterMouseMove() {
-      if (typeof (this.triggeredAfterMouseMove as any).cancel === 'function') {
-        (this.triggeredAfterMouseMove as any).cancel();
-      }
+      this.removeListener();
+      (
+        this.triggeredAfterMouseMove as DebouncedFunc<typeof CategoricalChartWrapper.prototype.triggeredAfterMouseMove>
```

Comment:
could this be typed where its defined instead of casted here?

Reply:
Okay